### DISCUSSION TITLE
[Question]<Can ESP32S2 support the camera of ov2640>

### DISCUSSION THREAD

**Discussion by liqijian101** [luc-github/ESP3D#756] [Questions]
What is your question ?
hi luc, I test the esp32s2 chip with esp-12 . It could not work . Did you test that?
thanks your .

**Comment by luc-github** [comment]
S2 is supported in refactoring branch which is not yet merged to V3
but I do not have S2 with camera, so I cannot says if it works, is there some devices available already ?

**Comment by liqijian101** [comment]
I designed an S2 module based on esp32Cam. Because I need more pins, is esp3d refactoring working, like uploading files.

**Comment by luc-github** [comment]
here latest tests
https://github.com/luc-github/ESP3D/issues/741

I did quick check on S2 and S3 but no camera on them

**Comment by liqijian101** [comment]
Another reason I tested esp32s2 was to add the ability to connect to the CH340 serial chip

**Comment by liqijian101** [comment]
If I use a Webcam, is the following setting necessary.
-DBOARD_HAS_PSRAM
-mfix-esp32-psram-cache-issue

**Comment by luc-github** [comment]
Use PSRAM with camera is recommended, so if you added PSRAM yes this is necessary, I did not check if S2 has different setting for camera as I do not have board, I am guessing they are identical to ESP32,
